src/sentry/utils/appleconnect/appstore_connect.py

```
@@ -79,127 +79,103 @@ def _get_next_page(response_json: Mapping[str, Any]) -> Optional[str]:
+        "&include=appStoreVersion,preReleaseVersion"
+        # sort newer releases first
+        "&sort=-uploadedDate"
+        # TODO: figure out if we actually can/want to download invalid or expired builds
+        # only include valid builds
+        "&filter[processingState]=VALID"
+        # and builds that have not expired yet
+        "&filter[expired]=false"
+    )
+    pages = _get_appstore_info_paged(session, credentials, url)
+
+    for page in pages:
+        included_relations = {}
+        for included in safe.get_path(page, "included"):
+            type = safe.get_path(included, "type")
+            id = safe.get_path(included, "id")
+            if type is not None and id is not None:
+                included_relations[(type, id)] = included
+
+        def get_related(relation: Any) -> Any:
+            type = safe.get_path(relation, "data", "type")
+            id = safe.get_path(relation, "data", "id")
+            if type is None or id is None:
+                return None
+            return included_relations.get((type, id))
+
+        for build in safe.get_path(page, "data"):
+            related_appstore_version = get_related(
+                safe.get_path(build, "relationships", "appStoreVersion")
+            )
+            related_prerelease_version = get_related(
+                safe.get_path(build, "relationships", "preReleaseVersion")
+            )
+            related_version = related_appstore_version or related_prerelease_version
+            if not related_version:
+                continue
+            platform = safe.get_path(related_version, "attributes", "platform")
+            version = safe.get_path(related_version, "attributes", "versionString")
+            build_number = safe.get_path(build, "attributes", "version")
+            if platform is not None and version is not None and build_number is not None:
```

here too i'm worried what happens with silently ignoring what i would expect to be malformed data by now